packages/commonwealth/server/routes/generateImage.ts

```diff
@@ -39,37 +41,89 @@ type generateImageResp = {
   imageUrl: string;
 };
 
+const generateImageWithRunware = async (description: string) => {
+  log.info('Generating image with Runware model: runware:100@1');
+  const response = await fetch('https://api.runware.ai/v1/images/generations', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${process.env.RUNWARE_API_KEY}`,
+    },
+    body: JSON.stringify([
+      {
+        taskType: 'imageInference',
+        taskUUID: uuidv4(),
+        positivePrompt: description,
+        width: 256,
+        height: 256,
+        model: 'runware:100@1',
+        numberResults: 1,
+        outputType: 'URL',
+        outputFormat: 'PNG',
+      },
+    ]),
+  });
+
+  if (!response.ok) {
+    const error = new Error(`Runware API error: ${response.statusText}`);
+    log.error('Runware API error: ' + error.message);
+    throw error;
+  }
+
+  const data = await response.json();
+  log.info('Successfully generated image with Runware');
+  return data.data[0].imageURL;
+};
+
+const generateImageWithOpenAI = async (description: string) => {
+  log.info('Generating image with OpenAI model: dall-e-2');
+  const response = await openai!.images.generate({
+    model: 'dall-e-2',
+    n: 1,
+    prompt: description,
+    size: '256x256',
+    response_format: 'url',
+  });
+
+  log.info('Successfully generated image with OpenAI');
+  return response.data[0].url;
+};
+
 const generateImage = async (
   models: DB,
   req: TypedRequestBody<generateImageReq>,
   res: TypedResponse<generateImageResp>,
 ) => {
   const { description } = req.body;
 
-  if (!openai) {
-    throw new ServerError('OpenAI not initialized');
-  }
-
   if (!description) {
     throw new AppError('No description provided');
   }
 
+  // Check OpenAI initialization if we're not using Runware
+  if (process.env.USE_RUNWARE !== 'true' && !openai) {
+    log.error('OpenAI not initialized and Runware not enabled');
+    throw new ServerError('OpenAI not initialized');
+  }
+
   let image;
   try {
-    const response = await openai.images.generate({
-      model: 'dall-e-2',
-      n: 1,
-      prompt: description,
-      size: '256x256',
-      response_format: 'url',
-    });
-
-    image = response.data[0].url;
+    log.info(
+      `Using image generation service: ${process.env.USE_RUNWARE ? 'Runware' : 'OpenAI'}`,
+    );
+    if (process.env.USE_RUNWARE === 'true') {
```

Comment:
Don't use `process.env` directly. Env var should always be accessed through a `config` object. Any existing usage of `process.env` is legacy and should not be considered the proper pattern.